solutions/CF Santa's Bot.mdx

```diff
@@ -0,0 +1,154 @@
+---
+id: cf-1279D
+source: CF
+title: Santa's Bot
+author: Kevin Sheng
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/72577)
+
+Clarification:
+The optimization the author uses is this property of fractions modulo $998244353$ is this:
+
+<Info>
+Say we had two fractions $x=\frac{a}{b}$ and $y=\frac{c}{d}$. Let $f$ be a function that calculates the "total" of a fraction (that is, the numerator times the modular inverse of the denominator). To find $f(x + y)$, we can simply calculate $f(x) + f(y)$. Thus, to find the total of the overall probability, we can just add up all the totals of all the smaller probabilities, keeping a running sum as we go.
+</Info>
```

Comment:
```suggestion
<Info>

Say we had two fractions $x=\frac{a}{b}$ and $y=\frac{c}{d}$. Let $f$ be a function that calculates the "total" of a fraction (that is, the numerator times the modular inverse of the denominator). To find $f(x + y)$, we can simply calculate $f(x) + f(y)$. Thus, to find the total of the overall probability, we can just add up all the totals of all the smaller probabilities, keeping a running sum as we go.

</Info>
```
a blank line is needed between markdown and jsx as mentioned [here](https://github.com/cpinitiative/usaco-guide/blob/master/docs/Content%20Documentation.md#using-custom-components) (i think this will fix the failed tests)